backend/src/data_provider/DataProvider.py

```diff
@@ -23,145 +18,64 @@ def start_data_provider():
-        previous_stop_times_dict = extractor.transform_stop_times_df_to_dict(stops_df, previous_stop_times_df, services_list)
-        next_stop_times_dict = extractor.transform_stop_times_df_to_dict(stops_df, next_stop_times_df, services_list)
-        print("FloydDataProvider: floyd extraction done")
-
-        # Pickle saving
-        nx.write_gpickle(floyd_graph, FloydDataPaths.floyd_graph.value)
-        nx.write_gpickle(kernelized_floyd_graph, FloydDataPaths.kernelized_floyd_graph.value)
-        save_pickle(distances, FloydDataPaths.distances.value)
-        save_pickle(day_to_services_dict, FloydDataPaths.day_to_services_dict.value)
-        save_pickle(current_stop_times_dict, FloydDataPaths.current_stop_times_dict.value)
-        save_pickle(previous_stop_times_dict, FloydDataPaths.previous_stop_times_dict.value)
-        save_pickle(next_stop_times_dict, FloydDataPaths.next_stop_times_dict.value)
-        save_pickle(routes_to_stops_dict, FloydDataPaths.routes_to_stops_dict.value)
-        save_pickle(adjacent_stops, FloydDataPaths.adjacent_stops.value)
-        save_pickle(exception_days, FloydDataPaths.exception_days.value)
-        stops_df.to_pickle(FloydDataPaths.stops_df.value)
-        routes_df.to_pickle(FloydDataPaths.routes_df.value)
-        stops_df_by_name.to_pickle(FloydDataPaths.stops_df_by_name.value)
-        stop_times_df.to_pickle(FloydDataPaths.stops_times_df.value)
-        print("FloydDataProvider: pickles saved")
-
     @staticmethod
     def load_update_date():
         load_property_from_config_json("update_date")
 
-    @staticmethod
-    def save_update_date(update_date):
-        save_property_to_config_json("update_date", update_date.strftime("%Y-%m-%d %H:%M:%S"))
-
-    def reparse_data(self):
-        new_update_date = self.downloader.get_last_update_time()
-
+    def process_data(self):
+        print("DataProvider: updating data")
         gtfs_zip_T, gtfs_zip_A = self.downloader.download_gtfs_static_data()
+        parsed_data_T = self.parser.parse(gtfs_zip_T)
+        parsed_data_A = self.parser.parse(gtfs_zip_A)
+        print("DataProvider: data parsed")
 
-        parser = Parser()
-        parsed_data_T = parser.parse(gtfs_zip_T)
-        parsed_data_A = parser.parse(gtfs_zip_A)
+        merged_data, service_id_offset = self.merger.merge(parsed_data_T, parsed_data_A)
+        print("DataProvider: data merged")
 
-        merger = Merger()
-        merged_data, service_id_offset = merger.merge(parsed_data_T, parsed_data_A)
+        corrected_data = self.corrector.correct(merged_data)
+        print("DataProvider: data corrected")
 
-        corrector = Corrector()
-        corrected_data = corrector.correct(merged_data)
+        extracted_data = self.extractor.extract(corrected_data)
+        print("DataProvider: data extracted")
 
         save_property_to_config_json("services", [[index for index in parsed_data_T.calendar_df.index],
```

Comment:
1. you can use syntax `list(parsed_data_T.calendar_df.index)` as well
2. we should store `service_id_offset` instead of `service_id`s